src/lightning/fabric/strategies/fsdp.py

```diff
@@ -253,6 +257,12 @@ def setup_environment(self) -> None:
         super().setup_environment()
         self._setup_distributed()
 
+        # if 'device_mesh' in the `_fsdp_kwargs` is provided as a tuple, update it into the `DeviceMesh` object here
+        if isinstance(self._fsdp_kwargs.get("device_mesh"), tuple):
+            from torch.distributed.device_mesh import init_device_mesh
+
+            self._fsdp_kwargs["device_mesh"] = init_device_mesh("cuda", self._fsdp_kwargs["device_mesh"])
```

Comment:
Since the tuple specification is a feature of Lightning, we should list the device_mesh parameter explicitly in the init args (see the docstring I added). The kwargs are for things that we pass down to FSDP directly.

So I suggest to set `self.device_mesh` and update this attribute here to the actual DeviceMesh 😃

Reply:
If `device_mesh` is separated from `kwarfs`, we will fail the check in `_init_sharding_strategy` in the `self.__init__`.

Reply:
Changed now. @awaelchli could you take a look again?